runtime/services/catalog/artifacts/sql/sql.go

```diff
@@ -24,13 +24,20 @@ func init() {
 	artifacts.Register(".sql", &artifact{})
 }
 
-func (r *artifact) DeSerialise(ctx context.Context, filePath, blob string) (*drivers.CatalogEntry, error) {
+func (r *artifact) DeSerialise(ctx context.Context, filePath, blob string, materializeDefault bool) (*drivers.CatalogEntry, error) {
 	name := fileutil.Stem(filePath)
 	// extract materialize option before sanitizing query as it will remove that comment
 	materialize := parseMaterializationInfo(blob)
 	if materialize == MaterializeInvalid {
 		return nil, errors.New("invalid materialize type")
 	}
+	if materialize == MaterializeInferred {
```

Comment:
this probably also need to match for MaterializeUnspecified as well..

Reply:
another way could be to pass in default to parseMaterializationInfo(blob) as well and resolve it there itself.

Reply:
Sorry, it's a mistake, it should only be for `MaterializeUnspecified` (`inferred` is never used, but IIRC was originally added to leave room for inferring materialization based on heuristics).